src/viam/app/data/client.py

```diff
@@ -290,10 +320,41 @@ async def bounding_box_labels_by_filter(self, filter: Optional[Filter] = None) -
         response: BoundingBoxLabelsByFilterResponse = await self._data_client.BoundingBoxLabelsByFilter(request, metadata=self._metadata)
         return response.labels
 
-    # TODO(RSDK-3637): Implement
-    async def data_capture_upload(self, metadata: UploadMetadata, sensor_contents: Optional[List[SensorData]] = None) -> None:
-        raise NotImplementedError()
+    async def data_capture_upload(self, metadata: UploadMetadata , sensor_contents: List[SensorData]) -> None:
+        """Upload binary or tabular sensor data.
+
+        Sync binary or tabular data collected on a robot through a specific component (i.e. a motor) along with the relevant metadata with
+        app.viam.com.
+
+        Args:
+            metadata (viam.app.proto.datasync.UploadMetaData): Metadata relating the data collected with the corresponding robot part. Must
```

Comment:
I appreciate the amount of detail you're putting into these docstrings, but I'm pretty sad that this info isn't available if I want to look at the type itself. Given that upload metadata is used for both upload methods, but which _parts_ of that data we care about are different, I think this is a good case for asking for just the data we need and then building an `UploadMetadata` object ourselves.

Reply:
I agree I think this would be a good change. The function signatures might get a little wonkier, but the benefit would be ease of use. It seems like the only place to get the definition of UploadMetadata is either through the .proto file that defines them or the .pyi generated from it - neither of which are super easy to decipher.